From 18b01a166ee25d50254038470f4355a5a2fe71d7 Mon Sep 17 00:00:00 2001
From: Kendall Willis <k-willis@ti.com>
Date: Mon, 1 Dec 2025 15:33:38 -0600
Subject: [PATCH 1/2] perf(ti): Decrease DDR speed for max power savings

Decrease the DDR clock speed by half in order to maximize power savings.
The clock is decreased to 200 MHz which indicates a 800 MT/s data
transfer rate.

Change-Id: I7b4860fdc34b815a51d9cdc88f9b0e7eb13b2401
Signed-off-by: Kendall Willis <k-willis@ti.com>
---
 plat/ti/k3/common/drivers/lpddr4/k3-ddrss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/ti/k3/common/drivers/lpddr4/k3-ddrss.c b/plat/ti/k3/common/drivers/lpddr4/k3-ddrss.c
index 475adbb00..f29491a90 100644
--- a/plat/ti/k3/common/drivers/lpddr4/k3-ddrss.c
+++ b/plat/ti/k3/common/drivers/lpddr4/k3-ddrss.c
@@ -293,7 +293,7 @@ static int ddrss_set_pll(unsigned long freq)
 	int ret = 0;
 
 	if (freq == ddrss.ddr_freq1)
-		ret = set_ddr_pll_div(4);
+		ret = set_ddr_pll_div(9);
 	else if (freq == 25000000)
 		ret = set_ddr_pll_div(79);
 	
-- 
2.34.1

